### DISCUSSION TITLE
Copy URL of active site with shift+cmd+c

### DISCUSSION THREAD

**Discussion by Kevoin** [zen-browser/desktop#1602] [Ideas]
It would be nice if we could use shift+cmd+c to copy the URL of the current site like arc. I know the shortcut currently open Dev-tools, maybe move Dev-tools to f12 instead (Like in arc and chrome)?

**Comment by noxan** [comment]
would love to see this, got so used to it on arc - hate when zen/firefox opens the developer console instead.

tried to build an [addon](https://addons.mozilla.org/en-US/firefox/addon/copy-url-to-clipboard/) to fix it but seems the shortcut is registered globally and there is no way to override it - settled with alternative for now. would love to see this in zen ❤️

**Comment by hjasond** [comment]
Adding my vote, I copy and paste URLs all the time and my muscle memory is cmd-shift-c. The ability to set this as a shortcut in Zen would be a huge quality of life improvement. Orion has also adopted the cmd-shift-c shortcut so we might as well join.

**Comment by B4ttleCat** [comment]
As a soon-to-be former Arc user, this one has broken me too. I do it without thinking now and I think it's the last thing holding me back from jumping permanently to Zen.

**Comment by lucasrabiec** [comment]
Dev tools are also on ctrl+shift+i (cmd+alt+i on mac I am not sure). I'm currently testing Zen (to use it instead of Arc) and it will be nice to have copy url shortcut in Zen.

**Comment by drfiresign** [comment]
Another vote for this shortcut. I didn't even think about not being able to do that in a browser, Arc had me spoiled so badly. I would love to see this functionality added.

**Comment by FKouhai** [comment]
another vote for this shortcut, It would be really cool if there was a way we could just customize it within the zen settings

**Comment by mkster** [comment]
need this too, has anybody found a workaround for now?

**Reply by rubaboo** [reply]
> need this too, has anybody found a workaround for now?

Here you go: <kbd>Alt</kbd>+<kbd>D</kbd> ,  <kbd>Ctrl</kbd>+<kbd>C</kbd>

**Reply by erutan** [reply]
cmd+L then cmd+c on macOS. :)

**Comment by erutan** [comment]
Is this just doable as a browser extension?

**Reply by rubaboo** [reply]
Absolutely. 
https://addons.mozilla.org/en-US/firefox/search/?q=Copy%20URL
A lot of requests here are coming from users completely new to the Firefox universe and unaware of the rich ecosystem of high-quality add-ons that it has.

**Reply by erutan** [reply]
It's been a while since Firefox was my primary browser on my main OS (it was my choice for many years). These days I generally use Arc for focused workspace centric stuff and Safari for casual browsing due to the power effeciency.

I wonder if it'd be useful for new users as part of the onboarding wizard flow to have some presets of good extensions that could be installed.  Given that this can be done with two sets of key presses I figured it had to be an extension heh.

**Reply by hjasond** [reply]
> Absolutely. https://addons.mozilla.org/en-US/firefox/search/?q=Copy%20URL A lot of requests here are coming from users completely new to the Firefox universe and unaware of the rich ecosystem of high-quality add-ons that it has.

Unfortunately, you cannot with cmd+shift+c because it is already a Firefox shortcut and cannot be changed with an extension. I have tried them all, if you have figured it out please let us know.

**Reply by erutan** [reply]
Interesting, it's not in the application's keyboard shortcuts.

https://support.mozilla.org/bm/questions/1381773 mentions https://addons.mozilla.org/en-US/firefox/addon/shortkeys/ there's also the option to change it with autoconfig if you don't want another extension.

You could also disable inspector if you don't use it for development. :)

**Reply by rubaboo** [reply]
No, I did not figure that out. Alt+D , Ctrl+C combo is good enough, and I have both of those widely used shortcuts firmly ingrained in my muscle memory. I would not use an extension just to reduce that to a single shortcut. I use MarkDownload, very flexible in what it can put into the clipboard. https://addons.mozilla.org/en-US/firefox/addon/markdownload/

**Reply by erutan** [reply]
opt+D, ctl+D, shit+D don't do anything on macOS in Zen.  cmd+D opens the save this page as a bookmark dialog. :)

I assume that alt-d pops you to the url bar and selects all in linux and/or windows?  The equivalent is cmd+L on macOS at least in Safari and Zen (it's hijacked in Arc), which is a little annoying if you're one hand on the trackpad/mouse and one on keys but works across the OS.  There's multiple ways to override that (BTT, Alfred, etc etc) to a better combination - one could even just set up a macro while in Firefox to run cmd+L then cmd+C.

It is nice just having it out of the box though.

**Reply by hjasond** [reply]
Even disabling the developer tools shortcut doesn't let you set it using one of these extensions. And yes, you can do the same thing with cmd+L and cmd+c I have a macro setup to hit those two commands when I type cmd+shift+c but its clunky. This would be a huge quality of life improvement in Zen who's muscle memory is set.

**Reply by erutan** [reply]
> Even disabling the developer tools shortcut doesn't let you set it using one of these extensions.

So autoconfig and shortkeys don't work?

> This would be a huge quality of life improvement in Zen who's muscle memory is set.

Yeah but it's built on Firefox and has Firefox people too.  I think having feature parity with Arc in a lot of ways is great, but out of the box changing Firefox keybinds (which have existed far longer than Arc has) doesn't feel right.

An issue that asked for more keybinds to be exposed would IMO be more appropriate - being able to change inspector  being bound to cmd/ctrl+shift+c would get around the need for kludgey workarounds while not disrupting people used to the existing way of doing things.

**Reply by hjasond** [reply]
> So autoconfig and shortkeys don't work?

correct

> An issue that asked for more keybinds to be exposed would IMO be more appropriate - being able to change inspector being bound to cmd/ctrl+shift+c would get around the need for kludgey workarounds while not disrupting people used to the existing way of doing things.

To be clear, all I want is the ability to customize keyboard shortcuts to match my workflow which is not possible in Zen today. Ideally this could be added to the Keyboard Shortcuts settings page. I have no interest in changing the default behavior, just the ability to customize the browser to function how I would like to use it.

**Reply by FKouhai** [reply]
yeah I agree, I think it would be a good addition to have that extra customization

**Reply by TobiasVonEssenQED** [reply]
I back that. Coming from Arc browser, I am very much missing this QoL feature.

**Comment by hjasond** [comment]
FYI for anyone following this discussion this feature was added in b3!

**Reply by zplawecki** [reply]
on Windows it doesn't work, this shortcut still opens dev tools

**Comment by mr-cheffy** [comment]
This existed for a while now! Closing.
